bin/octocatalog-diff

```diff
@@ -95,17 +153,57 @@ def all_diffs():
     TODO: Make this faster by just selecting a single node from each class we
     care about (not selecting all desktops/hozers for example)
     """
-    print('Setting up cache...')
+    info('Setting up cache...')
     setup_cache()
-    print('Getting hosts from puppetdb...')
+
+    info('Getting hosts from puppetdb...')
     hosts = get_hosts_from_puppetdb()
+
     # Get number of cores to parallelize across and create a worker pool of that size
     pool = Pool(os.cpu_count())
 
-    print('Running octocatalog-diff across all hosts...')
-    # Find any status codes that are non-zero (errors encountered in puppet runs)
-    returncode = max([ret for ret in pool.imap(diff, hosts)])
-    return returncode
+    info('Running octocatalog-diff across all hosts...')
+    run_output = pool.imap(diff, hosts)
```

Comment:
`imap_unordered`?

Reply:
It is nice that the hosts are ordered alphabetically e.g. for looking for the output of a particular host.

Reply:
Yeah, that's why I kept it ordered, having it alphabetical is pretty nice. I'm not sure otherwise what the difference is between the ordered and unordered versions or if `imap` ensures that the results are ordered, but it seems to be working reasonably well right now to return the hosts in sorted order by hostname so I haven't investigated any further.

Reply:
imap_unordered will usually run faster, but I don't think that is worth it here, and it seems silly to call imap_unordered and then sorted() afterwards.